agent/eventhandler/task_handler.go

```diff
@@ -116,6 +126,10 @@ func NewTaskHandler(ctx context.Context,
 		client:                    client,
 		minDrainEventsFrequency:   minDrainEventsFrequency,
 		maxDrainEventsFrequency:   maxDrainEventsFrequency,
+		cfg:                       cfg,
+		delay:                     delay,
+		eventFlowCtx:              nil,
```

Comment:
Is there a downside to initializing these? might be safer than intentionally passing around nils unless the nil value is useful for something?

Reply:
nil is not really useful yeah.. I can initialize eventFlowCtx to context.Background() but couldn't figure out what value to assign to eventFlowCtxCancel

Reply:
Since I've removed the nil initializations, also removed nil checks done for those variables in a couple of places